cabot/metricsapp/models/elastic.py

```diff
@@ -193,20 +193,35 @@ def _parse_series(self, series, series_name=None):
         :param series_name: the name for the series ('key1.key2. ... .metric')
         :return: (series, (timestamp, datapoint)) pairs
         """
-        for subseries in series:
-            # If there are no more aggregations, we've reached the metric
+
+        islist = isinstance(series, list)
+        i = 0
+
+        if not islist:
+            keyz = series.keys()
+        while i < len(series):
+            if not islist:
+                subseries = series[keyz[i]]
+                series_name = keyz[i]
```

Comment:
This will replace the current series_name with the key name--would prefer appending it like in line 217:
`series_name = '.'.join(filter(None, [series_name, keyz[i]]))`